src/main/java/net/masterthought/cucumber/ReportBuilder.java

```diff
@@ -147,27 +152,38 @@ private void copyResources(String resourceLocation, String... resources) {
         }
     }
 
-	private void generatePages(Trends trends) {
-		new FeaturesOverviewPage(reportResult, configuration).generatePage();
-		
-		for (Feature feature : reportResult.getAllFeatures()) {
-			new FeatureReportPage(reportResult, configuration, feature).generatePage();
-		}
-		
-		new TagsOverviewPage(reportResult, configuration).generatePage();
-		
-		for (TagObject tagObject : reportResult.getAllTags()) {
-			new TagReportPage(reportResult, configuration, tagObject).generatePage();
-		}
-
-		new StepsOverviewPage(reportResult, configuration).generatePage();
-		new FailuresOverviewPage(reportResult, configuration).generatePage();
-
-		if (configuration.isTrendsStatsFile()) {
-			new TrendsOverviewPage(reportResult, configuration, trends).generatePage();
-		}
-	}
-	
+    private List<AbstractPage> collectPages(Trends trends) {
```

Comment:
this method has been just refactored because of some memory issue - di you check it with the author?

Reply:
Hi, 

I've seen @sasjo refactored the method using the quick/dirty solution I posted in issue #682 . 

This pull request pretend to be a bettere solution

Reply:
The patch I submitted was all about being as small as possible to get it into production as quickly as possible. As long as this change doesn't reintroduce the two-step page generation that required all pages to be stored in memory I guess it will only reduce memory consumption even more.